src/main/scala/com/gu/octopusthrift/services/ArticleFinder.scala

```diff
@@ -1,32 +1,75 @@
 package com.gu.octopusthrift.services
 
 import com.gu.octopusthrift.models._
+import scala.util.{ Failure, Success, Try }
 
 object ArticleFinder extends Logging {
 
-  def findBodyText(bundle: OctopusBundle): Option[OctopusArticle] = {
-    val bodyTexts: Array[OctopusArticle] =
-      bundle.articles.filter(hasBodyTextForWebPublication)
-    getPriorityBodyText(bodyTexts)
+  private val BodyText = "Body Text"
+  private val PanelText = "Panel Text"
+  private val TabularText = "Tabular Text"
+  private val ForBoth = "b"
+  private val ForWeb = "w"
+  private val ForPrint = "p"
+
+  // Order of precedence for object types: body over panel over tabular
+  private val bodyTextObjects = List(BodyText, PanelText, TabularText)
+
+  // Sometimes, object types look like Body Text [Ruled] and we don't care about the part in brackets
+  private def cleanObjectType(objectType: String): String = objectType.split("\\[").head.trim
+
+  // Prefer articles that are labelled both, then web, then print
+  private def articlesForPreferredDestination(
+      articles: Map[String, Array[OctopusArticle]]
+  ): Array[OctopusArticle] = {
+    (Try(articles(ForBoth)), Try(articles(ForWeb)), Try(articles(ForPrint))) match {
+      case (Success(forBoth), _, _)                    => forBoth
+      case (Failure(_), Success(forWeb), _)            => forWeb
+      case (Failure(_), Failure(_), Success(forPrint)) => forPrint
+      case _                                           => Array.empty[OctopusArticle]
+    }
   }
 
-  private val bodyTextObjectTypes =
-    Seq("Body Text", "Tabular Text", "Panel Text")
-  private val forWebCodes = Seq("w", "b")
+  // Prefer Body Text, then Panel Text, then Tabular Text
+  private def articlesOfPreferredObjectType(
+      articles: Map[String, Array[OctopusArticle]]
+  ): Array[OctopusArticle] = {
+    val bodyText = Try(articles(BodyText))
+    val panelText = Try(articles(PanelText))
+    val tabularText = Try(articles(TabularText))
 
-  private def hasBodyTextForWebPublication(article: OctopusArticle): Boolean = {
-    bodyTextObjectTypes.contains(article.object_type) && forWebCodes.contains(
-      article.for_publication.toLowerCase()) && article.object_number == 1 // we want the first of any given object type
+    (bodyText, panelText, tabularText) match {
```

Comment:
`nit`I think I prefer this syntax of declaring the variables and then matching them. Either way it would be nice to be consistent.